packages/block-library/src/search/view.js

```diff
@@ -29,14 +32,25 @@ const { actions } = store(
 				const { isSearchInputVisible } = getContext();
 				return isSearchInputVisible ? '0' : '-1';
 			},
+			get isSearchInputVisible() {
+				const ctx = getContext();
+
+				// `ctx.isSearchInputVisible` is a client-side-only context value, so
+				// if it's not set, it means that it's an initial page load, so we need
+				// to return the value of `ctx.isSearchInputInitiallyVisible`.
+				if ( typeof ctx.isSearchInputVisible === 'undefined' ) {
+					return ctx.isSearchInputInitiallyVisible;
+				}
+				return ctx.isSearchInputVisible;
+			},
```

Comment:
I see this derived state is only being used inside the `openSearchInput()` action, and `isSearchInputInitiallyVisible` is not read elsewhere, which makes me think that its function can be absorbed by `ctx.isSearchInputVisible`, right? 🤔 

Note that we changed how the state and context behave, and now, their data is not updated from the server during client-side navigation. I feel it would be safe to read the initial value from `ctx.isSearchInputVisible` directly.

Reply:
I think that you're right! Good catch 👍

Reply:
In addition to being redundant now that we've changed the behavior of state and context on client-side navigations, it's also not really necessary as this fix was needed when the Search button is visible. 

In this PR we're intentionally hiding the Search button (because such UX makes little sense for Instant Search) so this branch of the code should never be reached.

Done in 77e86d21693d72c0da94a30ef78b1bfff3b911af